luatest/server.lua

```diff
@@ -431,6 +433,32 @@ local function is_header_set(headers, name)
     return false
 end
 
+-- Checking for the correctness of the passed `args` for the `fn` function
+local function is_malformed_args(fn, args)
+    local fn_nparams = debug.getinfo(fn).nparams
+    local fn_source = string.split(debug.getinfo(fn).source, '/')
+    local fn_details = string.format(
+        '%s:%s', fn_source[#fn_source], debug.getinfo(fn).linedefined) -- 'filename:line'
+
+    if args == nil then
+        if fn_nparams > 0 then
+            return string.format(MALFORMED_MSG, fn_details, fn_nparams, 'nil')
+        end
+    else
+        if #args ~= fn_nparams then
+            for k, _ in pairs(args) do
+                if type(k) ~= 'number' then
+                    local err = string.format(
+                        "bad argument type to %s", fn_details)
+                    error(err, 2)
+                end
+            end
+            return string.format(MALFORMED_MSG, fn_details, fn_nparams, #args)
+        end
+    end
+    return nil
+end
```

Comment:
I don't like the "procedure" style because:

1. Source code will look like this:

```lua
local function foo(a,b,c)
    foo_procedure(a)
    bar_procedure(b)
    ...
    -- Call stack state(or variables or etc) has been changed implicitly.
end
```

2. `server:exec` causes some errors, let's continue to follow this semantic rule:
```lua
function Server:exec(fn, args, options)
    ....
    if self.net_box == nil then
        error(...)
    end
    ...
    if next(other_ups) ~= nil then
        error(...)
    end
    ...
```

Reply:
Ok, I don't mind.